fuse/pathfs/pathfs.go

```diff
@@ -255,147 +242,100 @@ func (n *pathInode) SetInode(node *nodefs.Inode) {
 	n.inode = node
 }
 
-// Reread all client nodes below this node.  Must run outside the treeLock.
-func (n *pathInode) updateClientInodes() {
-	n.GetAttr(&fuse.Attr{}, nil, nil)
-	for _, ch := range n.Inode().FsChildren() {
-		ch.Node().(*pathInode).updateClientInodes()
+// getParent - returns a random parent of this pathInode and the name under this
+// parent.
+// The caller must hold pathLock.
+func (n *pathInode) getParent() (*pathInode, string) {
+	for k, _ := range n.parents {
```

Comment:
add a TODO somewhere: the parents map is only necessary for non-directory nodes. We could shortcut some indirections by having a parent pointer for directory nodes. (should benchmark first, though.)